front/src/common/authorization/hooks/useResourcesGrants.ts

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { mockedEditoastApi, type Grant, type ResourceType } from 'common/api/mock/mockEditoastApi';
+
+export type SubjectsGrantWithDetails = {
+  id: number;
+  type: string;
+  grant: Grant;
+  name: string;
+}[];
+
+/**
+ * provides the user grants on a list of resources
+ * @param payload the resource type and the resource ids
+ * @returns the user grants on the resources
+ */
+const useResourcesGrants = (payload: Partial<Record<ResourceType, number[]>>) => {
+  const hasResources = Object.entries(payload).some(([_, ids]) => ids && ids.length > 0);
```

Comment:
`ids` is always defined here
```suggestion
  const hasResources = Object.entries(payload).some(([_, ids]) => ids.length > 0);
  
```